File: lib/teamstory/connections/repository.ex
defmodule Teamstory.Connections.Repository do
  use Ecto.Schema
  import Ecto.Changeset

  schema "repositories" do
    field :uuid, Ecto.UUID
    field :avatar_url, :string
    field :base_url, :string
    field :deleted_at, :utc_datetime
    field :name, :string
    field :service, :string
    field :project_id, :id

    timestamps()
  end

  @doc false
  def changeset(repository, attrs) do
    repository
    |> cast(attrs, [:service, :avatar_url, :name, :base_url, :deleted_at, :project_id])
    |> Teamstory.Repo.generate_uuid()
    |> validate_required([:uuid, :service, :name, :project_id])
    |> unique_constraint([:project_id, :base_url])
  end
end

File: assets/src/models/repository.ts
export class Repository {
  public id: string = ''

  public name: string = ''

  public avatar_url?: string

  public base_url?: string

  public service: string = ''

  public static fromJSON(obj: Object): Repository {
    let item: Repository = Object.assign(new Repository(), obj)
    return item
  }
}
